File: 24.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
       if(!head || !head->next) return head;
       ListNode *next = head->next;
       head->next = swapPairs(next->next);
       next->next = head;
       return next;
    }
};

File: golang/24/24.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	nxt := head.Next
	head.Next = swapPairs(nxt.Next)
	nxt.Next = head
	return nxt
}
